src/test/java/teammates/test/cases/BaseTestCaseWithMinimalGaeEnvironment.java

```diff
@@ -14,6 +20,22 @@
 
     private LocalServiceTestHelper helper = new LocalServiceTestHelper();
 
+    /**
+     * Sets the environment of SystemProperty based on the URL.
+     */
+    @BeforeSuite
+    public void setUpEnvironment() {
+        try {
+            if (new URL(TestProperties.TEAMMATES_URL).getProtocol().equals("https")) {
+                SystemProperty.environment.set(SystemProperty.Environment.Value.Production);
+            } else {
+                SystemProperty.environment.set(SystemProperty.Environment.Value.Development);
+            }
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
```

Comment:
If we are not going to say anything here, we can add `throws MalformedURLException` to the method signature. TestNG will catch the error (stop the test) and report.